docs/operators/s3outputmodule.md

```diff
@@ -0,0 +1,197 @@
+
+### Multipart Upload Feature
+----------------------------
+Uploading the blocks of a file into S3 can be achieved by [multipart feature](http://docs.aws.amazon.com/AmazonS3/latest/dev/mpuoverview.html).
+
+Using Multipart Upload feature, we can upload a single large file in multiple smaller parts. In Multipart Upload process, each part is a portion of object's data. Each part can be uploaded independently. 
+After all parts of a file are uploaded successfully, Amazon S3 combines the parts as a single object.
+
+Please refer the [Java code](http://docs.aws.amazon.com/AmazonS3/latest/dev/llJavaUploadFile.html) for uploading file into Amazon S3 bucket using multipart feature.
+
+### Module
+----------
+Module provides a way to use a group of operator as a single entity in a application. Module will contain list of input ports, list of output ports, configuration properties.
+Module will populate internal components properties as required based on its properties.
+
+### Operators in S3OutputModule
+-----------------------------------------
+
+Following diagram illustrates the DAG in this module:
+
+![](images/s3output/s3outputmodule.png)
+
+-   ***S3InitateFileUploadOperator***
+    -   Initiate the upload for the file using **AmazonS3Client.initiateMultipartUpload(...)** method only if the number of blocks for a file is greater than 1. By successfully initiating the file, S3 returns a response of type **InitiateMultipartUploadResult**, which includes the **upload ID**, which is the unique identifier for the multipart upload. This **upload ID** must be included in each operation like upload the blocks, multipart upload completion request of a file.
+If the file has single block then the operator emits an empty string, this is an indication of not to use the multi-part feature for the downstream operators.
+    -   This operator emits the **(filemetadata, uploadId)** to **S3FileMerger** and **(filePath, metadata, uploadId)** to **S3BlockUpload**.
+
+-   ***S3BlockUploadOperator***
+    -   **S3BlockUpload** operator upload the blocks into S3 using different API's which depends on number of blocks of a file.
+If the file has single block then upload the block into S3 using **AmazonS3Client.putObject(...)** API. S3 returns a response of type **PutObjectResult** includes the **ETag**.
+If the file has more blocks then upload the block using **AmazonS3Client.uploadPart(...)** method. S3 returns a response of type UploadPartResult includes the **ETag**. This **ETag** value must be include in request to complete multipart upload.
+    -   **S3BlockUpload** emits **(path, ETag)** to **s3FileMerger**.
+
+-   ***S3FileMerger***
+    -   Complete multipart upload request using **AmazonS3Client.completeMultipartUpload(...)** and must include the **upload ID**, a list of both part numbers and corresponding **ETag** values. **S3FileMergerOperator** send the complete multi-part upload request to S3 once it have all the **part ETag's** of a file. 
```

Comment:
and must => . This call must
a list => and a list
**S3FileMergerOperator** send the complete multi-part upload request to S3 once it have all the **part ETag's** of a file. => **S3FileMergerOperator** sends .... once it has ....
{Here again, double check whether the actual operator name is S3FileMerger or S3FileMergerOperator}